feat(issues-platform): Start writing occurrences to platform

udameli opened the pull request:
Use system and project options set in https://github.com/getsentry/sentry/pull/45709

Fixes https://github.com/getsentry/sentry/issues/43975

udameli committed:
-wip

udameli committed:
write occurrences

udameli committed:
use options

udameli committed:
fix

udameli commented on src/sentry/event_manager.py:
```diff
@@ -2455,6 +2459,64 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+                        type=problem.type,
+                        issue_title=problem.title,
+                        subtitle=event.transaction,
+                        # TODO use this to set metadata.value="problem.desc"
```
perf issues have `value` field in metadata

udameli committed:
add a group type check

barkbarkimashark commented:
I know this bit of code is where we wanna avoid double-counting issue stats:
https://github.com/getsentry/sentry/blob/149907ee5df4b4513db2e813357518d477859bcb/src/sentry/issues/ingest.py#L61-L63

Gonna take a quick look for any others.

barkbarkimashark commented:
Hmm, not too sure if these TSDB models are used at all:
https://github.com/getsentry/sentry/blob/5bb33925a92b441d18a5c3aa84df2c37777cb395/src/sentry/event_manager.py#L2491

Need to double check whether we're actually querying this since we're not calling this function at all in issue platform.

udameli committed:
update

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli committed:
add more checks

udameli committed:
add fix

udameli committed:
simplify

udameli commented:
@barkbarkimashark I've also updated `send_issue_occurrence_to_eventstream` to skip consume if it's a perf issue with options enabled and added another check [here](https://github.com/getsentry/sentry/blob/5fd8242aca575c84314f1b6744be2ec1d2e9e471/src/sentry/tasks/post_process.py#L529-L532)

wedamija commented on src/sentry/tasks/post_process.py:
```diff
@@ -533,12 +533,23 @@ def post_process_group(
         "organizations:profile-blocked-main-thread-ppg", group_event.group.organization
     ):
         return
+
+    if (
+        group_event.group.issue_type.type_id
+        in get_group_types_by_category(GroupCategory.PERFORMANCE.value)
+        and options.get("performance.issues.send_to_issues_platform", False)
+        and group_event.project.get_option(
+            "sentry:performance_issue_send_to_issues_platform", False
+        )
+    ):
+        return
```
I think we want to remove this check. Passing `skip_consume=True` to `eventstream` means that we'll already skip sending these to `post_process_group`. But if we have this check here, then issue alerts will not fire for performance issues at all

udameli replied:
ah okay. removed!

wedamija commented on src/sentry/issues/ingest.py:
```diff
@@ -56,6 +57,13 @@ def save_issue_occurrence(
     group_info = save_issue_from_occurrence(occurrence, event, release)
```
I think that ideally we'd skip calling this as well when the option isn't enabled. We need to do this to avoid calling https://github.com/getsentry/sentry/blob/2ec9adaa218e4097caa36e456c160b7cbe906b94/src/sentry/issues/ingest.py#L204 which will update counts on the issue and cause double counting.

I wonder if the best option here might be to add logic into `save_issue_from_occurrence` that just fetches the group and returns it when the option is disabled. If the group doesn't exist, then it just returns None so that we don't process it any further. 

I guess there should never be a case where the group doesn't exist since it'd have already been created by the transaction, but probably worth just being safe?

udameli replied:
that makes sense. I've updated `save_issue_from_occurrence`

wedamija commented on src/sentry/event_manager.py:
```diff
@@ -2455,6 +2459,68 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+                            # TODO use this to set metadata.value="problem.desc"
+                            evidence_data={"value": problem.desc},
+                            evidence_display=[
+                                IssueEvidence(
+                                    name="parent_span_ids",
+                                    value=problem.parent_span_ids,
+                                    important=True,
+                                ),
+                                IssueEvidence(
+                                    name="cause_span_ids",
+                                    value=problem.cause_span_ids,
+                                    important=False,
+                                ),
+                                IssueEvidence(
+                                    name="offender_span_ids",
+                                    value=problem.offender_span_ids,
+                                    important=False,
+                                ),
```
Do we want the span ids to be displayed by default, or are they just used by the UI to render things?

If we just use them as data, then it might make sense to include these in `evidence_data` instead.

![Screen Shot 2023-03-14 at 9 19 51 AM](https://user-images.githubusercontent.com/6288560/224845645-293532a5-b262-4c38-9e52-481ca89bea24.png)
![Screen Shot 2023-03-14 at 9 23 50 AM](https://user-images.githubusercontent.com/6288560/224846196-e9ebb5bd-4dfc-4671-936c-0a3efd7a5666.png)

Based on the UI and email^, I think we want to have the actual span details in the `evidence_display`, and then the span ids in `evidence_data`

wedamija replied:
Although I'm not sure how generalizable that is to all perf issue types. But ideally, `evidence_display` has things we can display to the user, so that we can include them in email/slack/etc without having to customize for every new issue type.

udameli replied:
that makes sense. I update both evidence sections. I'm passing span ids because we look their descriptions on the client side but maybe that's not ideal

wedamija replied:
I'm wondering here if we should automatically translate the span ids to the span contents (just for `evidence_display`). That way it'll automatically display correctly. Then anything the FE client code needs to do with span ids can just work from `evidence_data`. I'm not totally sure if that makes sense?

This could be a thing where maybe we add `parent_spans, cause_spans, offender_spans` to the `PerformanceProblem`? And then we extract those from the transaction in the detector.
Since in the future, we won't actually be storing `PerformanceProblem` in nodestore at all, it'll just be used as an interface for you to send things to the issue platform.

Or maybe we should just emit `evidence_data` and `evidence_display` explicitly from the detectors. Then we'd be able to just shove it straight into evidence and it'd just work. I'm not sure if that makes sense here, but it means that each performance detector could more easily customize what they want to display.

udameli replied:
yeah I think `evidence_data` and `evidence_display` should be explicitly emitted from the detectors. So that I don't have to translate PerformanceProblem output into evidence because it'd be different for each detector. I'm going to add that change in 👍

udameli replied:
going back to this, we might need to go back to passing span ids in evidence display because we do some extra rendering customization like turning transaction name into a link and highlighting parameters in the url for http span description. adding some markup like that here is pretty cumbersome so I wonder if we need to go back to passing span ids and letting the client render the evidence sections as it needs
<img width="925" alt="Screenshot 2023-03-15 at 1 56 18 PM" src="https://user-images.githubusercontent.com/23648387/225401125-2a863236-c7e5-49fe-892d-f58c7f26fb2c.png">
<img width="1054" alt="Screenshot 2023-03-15 at 1 56 52 PM" src="https://user-images.githubusercontent.com/23648387/225401127-717eba1c-61b3-48eb-9e8a-807f2afe2972.png">

wedamija replied:
That's all basically UI customization, so in that case you're better off using the span ids from `evidence_data` and ignoring the `evidence_display` entirely. Then `evidence_display` can just be used for emails/slack (unless you also want to customize there)

udameli replied:
okay, then I'll keep everything as is 👍

udameli committed:
return early for save_issue_for_occurrence

udameli committed:
remove unnecessary check

udameli committed:
update evidence

wedamija commented on src/sentry/issues/ingest.py:
```diff
@@ -56,6 +57,13 @@ def save_issue_occurrence(
     if group_info:
         send_issue_occurrence_to_eventstream(event, occurrence, group_info)
+
+        if (
+            occurrence.type.type_id in get_group_types_by_category(GroupCategory.PERFORMANCE.value)
+            and options.get("performance.issues.send_to_issues_platform", False)
+            and event.project.get_option("sentry:performance_issue_send_to_issues_platform", False)
+        ):
+            return
```
Feels like we've repeating this a fair bit, does it make sense to wrap it in a utility function somewhere?

udameli committed:
compile evidence in the detector

udameli committed:
pull transaction name

udameli committed:
add a util function

udameli committed:
fix import

wedamija commented on src/sentry/utils/performance_issues/performance_problem.py:
```diff
@@ -15,6 +16,12 @@ class PerformanceProblem:
     # The actual bad spans
     offender_span_ids: Sequence[str]
+    # Evidence to be used for the group
+    # TODO make it required once all detectors have been migrated to platform
```
Just to add to the TODO: We can't make it required until we stop loading these from nodestore via `EventPerformanceProblem`, since there's legacy data in there that won't have these fields. So until we disable transaction based perf issues we'll need to keep this optional.

udameli replied:
added!

wedamija commented on src/sentry/utils/performance_issues/performance_problem.py:
```diff
@@ -43,6 +52,8 @@ def from_dict(cls, data: dict):
             data["parent_span_ids"],
             data["cause_span_ids"],
             data["offender_span_ids"],
+            data["evidence_data"],
+            data["evidence_display"],
```
We should do this via `data.get("evidence_data", [])` so that it doesn't break for existing perf problems we're fetching from nodestore.

udameli replied:
ah 👍 that might fix these 10000 failing tests

udameli replied:
updated!

wedamija commented on src/sentry/event_manager.py:
```diff
@@ -2455,6 +2459,52 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+                            type=problem.type,
+                            issue_title=problem.title,
+                            subtitle=event.transaction,
+                            # TODO need to set metadata.value="problem.desc"
```
Looks like I might have missed this as part of our ingestion schema. What does `metadata["value"]` get used for?

udameli replied:
it's used in the group stream. we generally show a query there 
<img width="886" alt="image" src="https://user-images.githubusercontent.com/23648387/225129764-92c6ec08-7729-4360-8569-17dd13dd7d6b.png">

udameli replied:
it's all the `SELECT`s and `GET /api/` in the screenshot

wedamija replied:
Ok, looks like I missed this into the `IssueOccurrence` model. Right now we have `title` - `N+1 Api Call` and `subtitle` - `GET /api/...`. But we don't have anything for location, which is currently extracted from the event.

This all happens here: https://github.com/getsentry/sentry/blob/b134ebb0d4cbe3d5a03cd633d4bbb1c29cbd8545/src/sentry/issues/ingest.py#L121-L139

Feels to me like `subtitle` should map to `metadata["value"]`, and we should have a new `location` value that maps to `metadata["location"]`, and we fall back to `event["location"]` if not present. Does that make sense to you?

udameli replied:
this is currently where the values are coming from in the UI 
<img width="629" alt="Screenshot 2023-03-14 at 5 53 53 PM" src="https://user-images.githubusercontent.com/23648387/225149172-015bbd99-bc7d-4908-8ac5-00507b8d1659.png">

for title, if there's no title in metadata, we'd default to pulling title from the group which is fine
for culprit, we read it from the group

so we don't really use location in this specific case

udameli replied:
it seems that in the UI we don't really use `subtitle` that's sent https://github.com/getsentry/sentry/blob/master/static/app/utils/events.tsx#L112

udameli replied:
> Feels to me like subtitle should map to metadata["value"], and we should have a new location value that maps to metadata["location"], and we fall back to event["location"] if not present. Does that make sense to you?

I think this can work. we're flipping culprit and metadata.value since the section right of the title is called a subtitle and the string below is called message but it should be fine I think

wedamija replied:
Ahh interesting. It looks like `culprit` just happens to be the same as `metadata["location"]` for perf issues, so I made that assumption.

Ok, so I think that we should just make `culprit` explicit as part of the occurrence then. So
 - `title` - `metadata["title"]`
 - `subtitle` - `metadata["value"]`
 - `culprit` - `culprit`

udameli replied:
yeah there are some fields that I don't think we use ever. 
👍 that works! I updated subtitle and can follow up with more changes after this PR

wedamija replied:
https://github.com/getsentry/sentry/pull/45809

udameli committed:
address comments

udameli committed:
add typing

udameli committed:
fix types for real

udameli committed:
more fixes

udameli committed:
remove changes to vscode settings

udameli committed:
clean up

udameli committed:
clean up

udameli committed:
add missings args

udameli committed:
send correct subtitle

udameli committed:
fixes

udameli committed:
fix type

udameli committed:
fix tests

udameli committed:
add missing args

udameli committed:
more test fixes

barkbarkimashark commented on src/sentry/event_manager.py:
```diff
@@ -2455,6 +2459,52 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+        project = event.project
+        event_id = event.event_id
+
+        use_platform = options.get("performance.issues.send_to_issues_platform", False)
+        if use_platform:
+            per_project_option = project.get_option(
+                "sentry:performance_issue_send_to_issues_platform", False
+            )
+            if per_project_option:
```
Question about this. How long are we planning to dual-write here by project? I ask because the backfill script I'm working on has some baked in assumptions that we'll be fully dual-writing for all orgs and projects on a certain day.

Is the plan to fully cut-over performance issues to issue platform something like this:
1. Merge this PR to master.
2. Test out dual-writing for sentry org 
3. Remove this check so we can fully dual-write for all org-projects (this would be the date I need for the backfill script to avoid ingesting dupes)
4. Run backfill script
5. Cut-over search queries and stats to query on the issue platform dataset instead of transactions table

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli committed:
add culprit

udameli committed:
handle performance groups in occurrence consumer

udameli commented on src/sentry/issues/occurrence_consumer.py:
```diff
@@ -262,9 +262,12 @@ def _process_message(
+                get_group_types_by_category(GroupCategory.PERFORMANCE.value)
+            )
+            if occurrence_data["type"] not in allowed_group_types or not features.has(
+                "organizations:profile-blocked-main-thread-ingest", organization
```
forgot to add perf issues to this check

udameli replied:
oh nevermind

udameli committed:
fix double hashing bug

udameli commented on src/sentry/issues/ingest.py:
```diff
@@ -67,6 +73,11 @@ def save_issue_occurrence(
+    # Do not hash fingerprints for performance issues because they're already
+    # hashed in save_aggregate_performance
+    # This check should be removed once perf issues are created through the platform
+    if occurrence_data["type"] in get_group_types_by_category(GroupCategory.PERFORMANCE.value):
+        return
```
I've noticed that we were double hashing which is why occurrences were never saved. this fixes that

wedamija replied:
I feel like this should maybe be based on `skip_group_processing`. Otherwise, when we switch over to primarily create via the issue platform we won't hash the fingerprints here.

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli commented:
I need to double check but somehow I see two issues created. Will report back

wedamija commented on src/sentry/event_manager.py:
```diff
@@ -2455,6 +2459,52 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+                        topic = settings.KAFKA_INGEST_OCCURRENCES
+                        cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+                        cluster_options = get_kafka_producer_cluster_options(cluster_name)
+                        producer = Producer(cluster_options)
+
+                        producer.produce(
+                            topic=topic,
+                            key=None,
+                            value=json.dumps(occurrence.to_dict()),
+                        )
+                        producer.flush()
```
I wasn't sure whether we were using this producer correctly for a high volume environment, so asked in sns. Turns out that we aren't using it correctly, and flushing on every message will slow things down a fair bit.

Slack thread is [here](https://sentry.slack.com/archives/CLTE78L73/p1678931612520379), but I'll summarise a bit: 

 - It's slightly better to cache the `Producer` and not instantiate it every time
 - Flushing on every message is bad, since it makes what was an async call into a sync one and will slow down processing
 - We should consider using the Arroyo `Producer` since it returns futures for each flush that we can wait on.
 - We'll need to keep the futures from each `produce` call around somewhere so that we can be sure they complete. Recommendations here are to store the futures in a global, then use `atexit` to await the futures.

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli committed:
fix a test

udameli committed:
use arroyo producer

udameli committed:
use arroyo producer properly

udameli committed:
set default options to False

udameli committed:
remove types for now

udameli committed:
store max futures limit in settings

udameli committed:
address comments

udameli commented on src/sentry/issues/producer.py:
```diff
@@ -0,0 +1,54 @@
+    global occurrence_producer_futures
+    occurrence_producer_futures.append(future)
+    if len(occurrence_producer_futures) >= 1:
+        future = occurrence_producer_futures.pop(0)
```
done

udameli commented on src/sentry/issues/producer.py:
```diff
@@ -0,0 +1,54 @@
+def track_occurrence_producer_futures(future: Future[BrokerValue[KafkaPayload]]) -> None:
+    global occurrence_producer_futures
+    occurrence_producer_futures.append(future)
+    if len(occurrence_producer_futures) >= 1:
```
updated 👍

udameli committed:
close producer to stop the script from hanging

udameli committed:
add types

barkbarkimashark commented:
> 

This issue is probably related to how we're doing search. Hopefully, this should fix it:
https://github.com/getsentry/sentry/pull/46016

barkbarkimashark commented on src/sentry/issues/producer.py:
```diff
@@ -0,0 +1,60 @@
+
+
+def produce_occurrence_to_kafka(occurrence: IssueOccurrence) -> None:
+    payload = KafkaPayload(None, json.dumps(occurrence.to_dict()).encode("utf-8"), [])
+    occurrence_producer = get_occurrence_producer()
+    future = occurrence_producer.produce(Topic(settings.KAFKA_INGEST_OCCURRENCES), payload)
+
+    track_occurrence_producer_futures(future)
```
Do we need to gate this function for when `handle_occurrence_producer()` gets called on app exit? Is it possible futures keep getting appended to `occurrence_producer_futures` as the app is exiting?

wedamija replied:
I don't think so, once `atexit` happens we're basically exiting the program, and so we shouldn't be producing anything extra.

wedamija commented on src/sentry/issues/utils.py:
```diff
@@ -0,0 +1,15 @@
+) -> bool:
+    return (
+        occurrence.type.type_id in get_group_types_by_category(GroupCategory.PERFORMANCE.value)
+        and options.get("performance.issues.send_to_issues_platform", False)
+        and project.get_option("sentry:performance_issue_send_to_issues_platform", False)
```
Are these backwards? This function should return `True` when we should skip group processing, so should it be:
`is_performance and (not option_1 and not option_2)`?

I think that when this is `False` we also need to make sure that we're not creating groups via the old method.

udameli replied:
yeah I think I need two separate flags here - to send occurrences to the platform (this will be tied to skipping group processing) and create a group. So we can first enable writing occurrences and then separately roll out group creation.

udameli committed:
also check for options before fingerprint hashing

udameli committed:
add types

udameli committed:
add group creation options

wedamija commented on src/sentry/issues/producer.py:
```diff
@@ -0,0 +1,61 @@
+    if occurrence_producer is None:
+        cluster_name = settings.KAFKA_TOPICS[settings.KAFKA_INGEST_OCCURRENCES]["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        producer_config.pop("compression.type")
+        producer_config.pop("message.max.bytes")
```
Just a reminder for myself to look into these

wedamija replied:
Checked with SNS, these config vals don't matter

wedamija commented on src/sentry/issues/utils.py:
```diff
@@ -0,0 +1,32 @@
+
+
+def can_create_group(
+    ocurrence: IssueOccurrence, options: Mapping[str, Any], project: Project
```
Why do we pass options in here instead of importing?

udameli replied:
oh I thought passing in a dependency explicitly is better but yeah fixed it to use the import instead

wedamija reviewed (commented):
I think we're pretty close here. One thing we still need to do is skip these when `can_create_group` is True

https://github.com/getsentry/sentry/blob/a2dacb3cd7ef192b09c5ead84b6c8f12869a9fa4/src/sentry/event_manager.py#L2485
https://github.com/getsentry/sentry/blob/a2dacb3cd7ef192b09c5ead84b6c8f12869a9fa4/src/sentry/event_manager.py#L2487
https://github.com/getsentry/sentry/blob/a2dacb3cd7ef192b09c5ead84b6c8f12869a9fa4/src/sentry/event_manager.py#L2490

udameli committed:
fix skip_consume check

udameli committed:
import options

udameli committed:
add more cheks

udameli commented on src/sentry/event_manager.py:
```diff
@@ -863,6 +867,11 @@ def _get_or_create_release_many(jobs: Sequence[Job], projects: ProjectsMapping)
+        if options.get("performance.issues.create_issues_through_platform", False) and projects[
+            job["project_id"]
+        ].get_option("sentry:performance_issue_create_issue_through_platform", False):
+            continue
```
not sure how this is going to work when there is a mix of n+1 db queries and other types in outputted performance problems. maybe it makes sense to turn group creation through platform on for all issue types at the same time

wedamija replied:
It might be easier. Another option could be to build group creation into the policy layer, but I'm not sure if it's worth it since it's just temporary for this transition.

udameli replied:
I was adding a check in the wrong place. resolving this since it's no longer applicable

udameli commented on src/sentry/event_manager.py:
```diff
@@ -1048,9 +1057,13 @@ def _create_kwargs(job: Union[Job, PerformanceJob]) -> dict[str, Any]:
+        ) and project.get_option("sentry:performance_issue_create_issue_through_platform", False):
+            return
+
+        job["environment"] = Environment.get_or_create(project=project, name=job["environment"])
```
same concern here

barkbarkimashark reviewed (changes requested):
Please hold off on merging this until I rebuild the search_issues dataset with the correct primary key.

udameli committed:
add proper checks

udameli commented on src/sentry/event_manager.py:
```diff
@@ -2327,7 +2338,14 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
+            performance_problems = [
+                problem
+                for problem in all_performance_problems
+                if not write_occurrence_to_platform(problem, project)
```
oh yeah you're right. fixed it!

udameli commented on src/sentry/event_manager.py:
```diff
@@ -863,6 +867,11 @@ def _get_or_create_release_many(jobs: Sequence[Job], projects: ProjectsMapping)
+        if options.get("performance.issues.create_issues_through_platform", False) and projects[
+            job["project_id"]
+        ].get_option("sentry:performance_issue_create_issue_through_platform", False):
+            continue
```
> We don't need to filter like this in _get_or_create_release_many and _get_or_create_environment_many - transactions should create those regardless of whether they create groups. We just need to filter in _get_or_create_group_environment_many and _get_or_create_group_release_many.

oops, added correct checks

> It might actually be enough to just filter in _save_aggregate_performance, since those other functions operate only on the groups provided.

I've added a check just in case to be sure

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli committed:
add tests

udameli committed:
Merge branch 'master' into du/feat/start-writing-perf-issues-to-platform

udameli commented on src/sentry/issues/producer.py:
```diff
@@ -0,0 +1,61 @@
+occurrence_producer_futures: Deque[Future[BrokerValue[KafkaPayload]]] = deque()
+
+
+def track_occurrence_producer_futures(future: Future[BrokerValue[KafkaPayload]]) -> None:
```
added!

udameli committed:
fix types

wedamija commented on src/sentry/issues/ingest.py:
```diff
@@ -39,7 +41,11 @@
+    # Do not hash fingerprints for performance issues because they're already
+    # hashed in save_aggregate_performance
+    # This check should be removed once perf issues are created through the platform
+    if can_create_group(occurrence_data, event.project):
```
`IssueOccurrenceData` is a dict, we'll need to handle that type in `can_create_group` as well

wedamija reviewed (commented):
I think this is in a good state now, basically ready to merge once that last fix is in, and we've rebuilt the table.

udameli committed:
handle dict case

wedamija committed:
fix test

wedamija committed:
fix types

wedamija committed:
make sure we keep ingesting non-performance issue types

wedamija commented on src/sentry/event_manager.py:
```diff
@@ -1154,12 +1159,13 @@ def _get_or_create_group_release(
-                environment=environment,
-                datetime=event.datetime,
-            )
+            if not can_create_group(group_info.group, group_info.group.project):
```
I think this check is causing problems - We actually use `_get_or_create_group_release` in the issues ingest pipeline as well, and separately it's causing problems with errors. I think it's fine to rely on not adding the groups in `_save_aggregate_performance` instead, going to try removing

udameli replied:
oh that's what it was. yeah I saw all these failures

wedamija committed:
another fix

udameli committed:
fix test

udameli committed:
fix missing evidence properties

wedamija committed:
temporarily log durations

wedamija committed:
try again

wedamija committed:
make more verbose

wedamija committed:
make sure producer is closed

wedamija committed:
print when we produce to kafka to debug

wedamija committed:
love too computer

mattgauntseo-sentry commented on Makefile:
```diff
@@ -131,7 +131,10 @@ test-python-ci:
 		--ignore tests/sentry/region_to_control/test_region_to_control_kafka.py \
-		--cov . --cov-report="xml:.artifacts/python.coverage.xml"
+		--cov . --cov-report="xml:.artifacts/python.coverage.xml" \
+		--durations=0 \
```
Not sure if you meant to keep this in, but if we we have this flag defined here: https://github.com/getsentry/sentry/blob/master/.github/actions/setup-sentry/action.yml#L77

wedamija replied:
We're debugging some stuff at the moment

wedamija committed:
maybe fix

wedamija committed:
*shrug*

wedamija committed:
actually fix it

wedamija committed:
cleanup

udameli committed:
fix key value error

wedamija committed:
don't need this

wedamija reviewed (approved):
lgtm. @barkbarkimashark let us know when we're safe to merge here

barkbarkimashark reviewed (approved):
sorry, didn't look too closely at the actual code changes but we should be gtg for dual writes.
